flytestdlib/resolver/k8s_resolver_test.go

```diff
@@ -0,0 +1,177 @@
+package resolver
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/serviceconfig"
+	"gotest.tools/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	testclient "k8s.io/client-go/kubernetes/fake"
+)
+
+func parseTarget(target string) resolver.Target {
+	u, err := url.Parse(target)
+	if err != nil {
+		panic(err)
+	}
+
+	return resolver.Target{
+		URL: *u,
+	}
+}
+
+type fakeConn struct {
+	cmp   chan struct{}
+	found []string
+}
+
+func (fc *fakeConn) UpdateState(state resolver.State) error {
+	for _, a := range state.Addresses {
+		fc.found = append(fc.found, a.Addr)
+	}
+	fc.cmp <- struct{}{}
+	return nil
+}
+
+func (fc *fakeConn) ReportError(e error) {
+	log.Println(e)
+}
+
+func (fc *fakeConn) ParseServiceConfig(_ string) *serviceconfig.ParseResult {
+	return nil
+}
+
+func (fc *fakeConn) NewAddress(_ []resolver.Address) {}
+
+func (*fakeConn) NewServiceConfig(serviceConfig string) {
+	fmt.Printf("serviceConfig: %s\n", serviceConfig)
+}
+
+func TestBuilder(t *testing.T) {
+	k8sClient := testclient.NewSimpleClientset()
+	builder := NewBuilder(context.Background(), k8sClient, "test")
+	fc := &fakeConn{
+		cmp: make(chan struct{}),
+	}
+	k8sResolver, err := builder.Build(parseTarget("test://flyteagent.flyte.svc.cluster.local:8000"), fc, resolver.BuildOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Make sure watcher is started before we create the endpoint
+	time.Sleep(5 * time.Second)
```

Comment:
Is it possible to make the waiting time shorter? This increases build time.

Reply:
I set it to 2 now